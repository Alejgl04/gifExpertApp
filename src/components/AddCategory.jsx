import { useState } from 'react'

export const AddCategory = () => {

  const [inputValue, setinputValue] = useState('')
  
  const onInputChange = ({ target }) => {
    setinputValue( target.value )
  }

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(event);
  }

  return (

    <form onSubmit={ ( event ) => onSubmit(event) }>
      <input 
        type="text"
        placeholder="Search gifs"
        value={inputValue}
        onChange={ (event) => onInputChange(event)}
      />
    </form>

  )
}
