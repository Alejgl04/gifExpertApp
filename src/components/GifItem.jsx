
export const GifItem = ({ title, url }) => {

  return (
    <div className="col-md-3">

    <div className="card">
      <img src={url} className="card-img-top custom__height" alt={title}/>
      <div className="card-body">
      <h6 className="card-title">{title}</h6>
      </div>
    </div>
    </div>
  )
}
