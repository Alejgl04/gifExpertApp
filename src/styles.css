* {
  font-family: Helvetica, Arial, sans-serif;
  /* background-color: rgb(234, 234, 234); */
}

body {
  background: linear-gradient(-45deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
  background-size: 400% 400%;
  animation: gradient 15s ease infinite;
  height: 100vh;
}

.title__form {
  color: #fff;
}
  
@keyframes gradient {
  0% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}  


button {
  padding: 0.5rem;
  margin-top: 0.5rem;
  background-color: #04AA6D;
  color: white;
  border: none;
  cursor: pointer;
  border-radius: 5px;
  margin-left: 5rem;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h1 {
  margin: 0;
  font-size: 2rem;
  margin-right: 2rem;
}

h2 {
  font-size: 1.5rem;
}


h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}

.center-form {
  display: flex;
  margin-top: 2.5rem;
  justify-content: center;

}

.custom__height {
  height: 250px;
}

@media screen and (max-width: 600px) {
  .center-form {
    display: block;
  }
}