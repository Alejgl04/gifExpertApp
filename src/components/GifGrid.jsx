import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/gift/useFetchGifs';


export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs( category );
  
  return (
    <>
      {
        isLoading && ( <h2 className='text-center text-white'>Loading...</h2> )
      }
      <h3 className='text-center text-white'>{category}</h3>
      
      <div className="container text-center">
      <div className="row align-items-start">
        {
          images.map( (image) => (
            <GifItem 
              key={image.id}
              {...image}
            />
          ))
        }
      </div>
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}