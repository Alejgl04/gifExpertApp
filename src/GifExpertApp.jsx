import { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Fifa 2023']);

  const onAddCategory = ( newCategory ) => {

    if ( categories.includes(newCategory) ) return; 

    setCategories([ newCategory, ...categories ]);
  }

  return (
    
    <>
      <div className='center-form'>
      <h1>GifsExpertApp..</h1> 
        <AddCategory 
          onNewCategory={ event => onAddCategory(event)}          
        />
      </div>

      <hr /> 
      { 
        categories.map( (category) => (
          <GifGrid 
            key={category} 
            category={category} 
          />
        ))
      }
    </>
  )
}
