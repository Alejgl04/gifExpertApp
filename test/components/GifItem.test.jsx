import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';


describe('Testing on GifItem', () => {

  const title = 'Spiderman';
  const url   = 'https://spiderman.com/spiderman.jpg';

  test('should be match snapshop', () =>{

    const { container } = render( <GifItem title={ title } url={ url }/>)
    expect( container ).toMatchSnapshot();
    
  });

  test('it must show the image with url and alt in the image', () =>{

    render( <GifItem title={ title } url={ url }/>)

    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
    
  });

  test('it must show the title component', () =>{

    render( <GifItem title={ title } url={ url }/>)
    expect( screen.getByText(title) ).toBeTruthy();
    
  });

});