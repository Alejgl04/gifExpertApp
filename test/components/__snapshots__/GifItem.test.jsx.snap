// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on GifItem should be match snapshop 1`] = `
<div>
  <div
    class="col-md-3"
  >
    <div
      class="card"
    >
      <img
        alt="Spiderman"
        class="card-img-top custom__height"
        src="https://spiderman.com/spiderman.jpg"
      />
      <div
        class="card-body"
      >
        <h6
          class="card-title"
        >
          Spiderman
        </h6>
      </div>
    </div>
  </div>
</div>
`;
